src/main/java/com/cleanroommc/groovyscript/compat/mods/advancedrocketry/BaseMultiblockRegistry.java

```diff
@@ -0,0 +1,69 @@
+package com.cleanroommc.groovyscript.compat.mods.advancedrocketry;
+
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import net.minecraft.item.ItemStack;
+import zmaster587.libVulpes.interfaces.IRecipe;
+
+import java.util.*;
+
+public abstract class BaseMultiblockRegistry extends BaseRegistry {
+
+    @MethodDescription(example = @Example(priority = 2000, commented = true))
+    public void removeAll() {
+        List<IRecipe> recipes = getRecipeList();
+        recipes.forEach(this::addBackup);
+        recipes.clear();
+    }
+
+    @MethodDescription(type = MethodDescription.Type.QUERY)
+    public SimpleObjectStream<IRecipe> streamRecipes() {
+        List<IRecipe> recipes = getRecipeList();
+        return new SimpleObjectStream<>(recipes)
+            .setRemover(this::removeRecipe);
+    }
+
+    public static abstract class RecipeBuilder extends BaseRegistry.RecipeBuilder {
+
+        // GrS cannot recognize these annotations on the builder class itself, so we have to redefine the fields
+        @Property(valid = @Comp(type = Comp.Type.GTE, value = "1"), value = "groovyscript.wiki.advancedrocketry.power.value")

```

Comment:
That sucks. So basically what youre saying is my code overwrites the variables in the superclass rather than pointing to the same variable? (I didn't actually expect "pointing to the same variable" worked but it compiled so.)

Reply:
I've just discovered `needsOverride()`. I'm not sure how exactly it works, but seems like it should be helpful.